shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedProcessorSyncServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.redhat.service.smartevents.shard.operator.v2;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.smartevents.infra.v2.api.models.ConditionStatus;
+import com.redhat.service.smartevents.infra.v2.api.models.OperationType;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ConditionDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ProcessorDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ProcessorStatusDTO;
+import com.redhat.service.smartevents.shard.operator.v2.converters.ProcessorStatusConverter;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedProcessor;
+
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_PROCESSOR_DELETED_NAME;
+
+@ApplicationScoped
+public class ManagedProcessorSyncServiceImpl implements ManagedProcessorSyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedProcessorSyncServiceImpl.class);
+
+    @Inject
+    ManagerClient managerClient;
+
+    @Inject
+    ManagedProcessorService managedProcessorService;
+
+    @Override
+    public void syncManagedProcessorWithManager() {
+        managerClient.fetchProcessorsForDataPlane()
+                .onItem()
+                .invoke(this::processDelta)
+                .subscribe().with(
+                        success -> LOGGER.debug("Successfully processed ManagedProcessor deltas"),
+                        error -> LOGGER.error("Failed to process ManagedProcessor deltas"));
+    }
+
+    @Override
+    public void syncManagedProcessorStatusBackToManager() {
+        managerClient.fetchProcessorsForDataPlane()
+                .onItem()
+                .transform(this::transformToProcessorStatus)
+                .invoke(this::notifyProcessorStatus)
+                .subscribe().with(
+                        success -> LOGGER.debug("Successfully sync ManagedProcessor status with Manager"),
+                        error -> LOGGER.error("Failed to sync ManagedProcessor status with Manager", error));
+    }
+
+    private void processDelta(List<ProcessorDTO> processorDTOList) {
+        for (ProcessorDTO processorDTO : processorDTOList) {
+            if (processorDTO.getOperationType() == OperationType.DELETE) {
+                managedProcessorService.deleteManagedProcessor(processorDTO);
+            } else {
+                managedProcessorService.createManagedProcessor(processorDTO);
+            }
+        }
+    }
+
+    private List<ProcessorStatusDTO> transformToProcessorStatus(List<ProcessorDTO> processorDTOList) {
```

Comment:
@vaibhavjainwiz I think this method is called for every `ProcessorDTO` we receive from the Control Plane? If yes, then we need to reconsider the placement of the `managedProcessorService.fetchAllManagedProcessors()` call as it will being called `N` times, where `N` is the number of `ProcessorDTO` received.

Reply:
no, it just called once with `List<ProcessorDTO>`.

Reply:
Thanks for clarifying @vaibhavjainwiz 👍